ext/spl/spl_directory.stub.php

```diff
@@ -0,0 +1,285 @@
+<?php
+
+class SplFileInfo
+{
+    public function __construct(string $file_name) {}
+
+    /** @return string */
+    public function getPath() {}
+
+    /** @return string */
+    public function getFilename() {}
+
+    /** @return string */
+    public function getExtension() {}
+
+    /** @return string */
+    public function getBasename(string $suffix = UNKNOWN) {}
+
+    /** @return string|false */
+    public function getPathname() {}
+
+    /** @return int|false */
+    public function getPerms() {}
+
+    /** @return int|false */
+    public function getInode() {}
+
+    /** @return int|false */
+    public function getSize() {}
+
+    /** @return int|false */
+    public function getOwner() {}
+
+    /** @return int|false */
+    public function getGroup() {}
+
+    /** @return int|false */
+    public function getATime() {}
+
+    /** @return int|false */
+    public function getMTime() {}
+
+    /** @return int|false */
+    public function getCTime() {}
+
+    /** @return string|false */
+    public function getType() {}
+
+    /** @return bool */
+    public function isWritable() {}
+
+    /** @return bool */
+    public function isReadable() {}
+
+    /** @return bool */
+    public function isExecutable() {}
+
+    /** @return bool */
+    public function isFile() {}
+
+    /** @return bool */
+    public function isDir() {}
+
+    /** @return bool */
+    public function isLink() {}
+
+    /** @return string|false */
+    public function getLinkTarget() {}
+
+    /** @return string|false */
+    public function getRealPath() {}
+
+    /** @return SplFileInfo */
+    public function getFileInfo(string $class_name = SplFileInfo::class) {}
+
+    /** @return SplFileInfo|null */
+    public function getPathInfo(string $class_name = UNKNOWN) {}
+
+    /**
+     * @param resource $context
+     * @return SplFileObject
+     */
+    public function openFile(string $open_mode = 'r', bool $use_include_path = false, $context = UNKNOWN) {}
```

Comment:
> I feel like probably we should fix this to also pass through the mode, the include path flag and the context...

By passing through the params, you mean to only pass them through ZPP + get rid of the hard-coded `"r"`? Or should I also call the constructor with all the params instead of just 2?

Reply:
For the purpose of this change, it would be enough to always perform the zpp call so the signature doesn't lie. But generally, yes, I think the constructor should be invoked with all 4 arguments instead of just 2.

Reply:
OK, I've just committed my attempt to fix ZPP + pass the received file open mode.